### PR TITLE
fixed bug

### PR DESCRIPTION
**Pull request by kialj876**
Signed-off-by: Kial Jinnah <kialj876@gmail.com>

*Issue #:* /bcgov/entity# OPS

*Description of changes:*
- *par value* and *currency* are not required in the schema: fixed failure in colin-api when they are not provided

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Review by rarmitag — APPROVED** [review]

File: colin-api/src/colin_api/models/shares.py [hunk 1]
```diff
@@ -261,17 +261,20 @@ def create_share_class(cls, cursor, event_id: str, corp_num: str, class_id: int,
...
                 class_id=class_id,
                 event_id=event_id,
-                currency=class_dict['currency'],
+                currency=currency,
                 has_max_share='N' if class_dict['hasMaximumShares'] else 'Y',
                 qty=class_dict['maxNumberOfShares'],
                 has_spec_rights='Y' if class_dict['hasRightsOrRestrictions'] else 'N',
                 has_par_value='Y' if class_dict['hasParValue'] else 'N',
-                par_value=class_dict['parValue'],
+                par_value=class_dict['parValue'] if class_dict['hasParValue'] else '',
```
Comment (thorwolpert) [inline colin-api/src/colin_api/models/shares.py:277]: same ```par_value = class_dict.get('parValue', '')```

**Review by thorwolpert — COMMENTED** [review]
suggest a shorter set of code?

File: colin-api/src/colin_api/models/shares.py [hunk 2]
```diff
@@ -261,17 +261,20 @@ def create_share_class(cls, cursor, event_id: str, corp_num: str, class_id: int,
             """
         )
         try:
+            currency = ''
+            if 'currency' in class_dict.keys() and class_dict['currency']:
+                currency = class_dict['currency']
```
[the remark this answers is not in the document]
Comment (kialj876) [reply colin-api/src/colin_api/models/shares.py:266]: HAHA thank you, I knew when I was writing this that there was a better way and it was bothering me. I've been in JS code on the front end too much lately

**Comment by argush3** [comment]
Was talking to Kial about this and it does seem like `currency` and `parValue` are required by the share structure json schema and it's also required by the legal api validation.  do you have any thoughts on this @thorwolpert?  i.e. is it ok to relax the constraints to allow no values for `currency` and `parValue` in the colin api?

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**